frontend/packages/console-dynamic-plugin-sdk/src/api/api-types.ts

```diff
@@ -43,3 +44,18 @@ export type UseK8sWatchResources = <R extends ResourcesObject>(
 export type UseResolvedExtensions = <E extends Extension>(
   ...typeGuards: ExtensionTypeGuard<E>[]
 ) => [ResolvedExtension<E>[], boolean, any[]];
+
+/* Horizontal Nav Types */
+export type NavPage = {
+  href?: string;
+  path?: string;
+  name?: string;
+  nameKey?: string;
```

Comment:
We shouldn't need these `key` properties in the API.

Reply:
```
            <Link
              to={`${baseURL.replace(/\/$/, '')}/${href}`}
              data-test-id={`horizontal-link-${nameKey || name}`}
            >
              {nameKey ? t(nameKey) : name}
            </Link>
```
`nameKey` is strictly for translation purposes. Should we perhaps expose `nameKey` as `name`? WDYT? But there would be no way to enforce a translation namespace.

Reply:
Translation should be left to the parent to pass in a proper translated string.